source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -286,6 +288,36 @@ void AuthenticatorImpl::handleGoodJwt(bool cache_hit) {
     }
   }
 
+  // Copy JWT Claim to Header
+  if (provider.claim_to_headers_size() != 0) {
+    for (const auto& header_and_claim : provider.claim_to_headers()) {
+      if (!header_and_claim.claim_name().empty() && !header_and_claim.header_name().empty()) {
```

Comment:
not need this "emtpy" check,  validation will fail if any of them is empty.

Reply:
If we go ahead without this check and any of the value is empty, it could create issues while fetching claim or copying the value to the header. It's safe to add this check IMO

Reply:
envoy filter_config should have been validated before passing it to the jwt_authn filter to use.   per your validation rules, an empty header or claim will fail the config,  the config will not be used.  That is why you don't need to check here.

Reply:
understood, thanks for the clarification